service/matching/matching_engine.go

```diff
@@ -2221,8 +2224,20 @@ func (e *matchingEngineImpl) DispatchNexusTask(ctx context.Context, request *mat
 	}
 
 	taskID := uuid.New()
+
+	// Buffer the deadline so we can still respond with timeout if we hit the deadline while dispatching
+	ctx, cancel := contextutil.WithDeadlineBuffer(ctx, matching.DefaultTimeout, NexusTimeoutBuffer)
+	defer cancel()
+
 	resp, err := pm.DispatchNexusTask(ctx, taskID, request)
 
+	if ctx.Err() != nil {
```

Comment:
Hmm... now that I think of it, this is before we even give a poll to be matched with a task. This should be a context deadline error that will be translated to an internal error.

Reply:
Explain in a comment that this request comes in with a buffer of 500 millis and should be enough to handle a request to dispatch internally to the partition manager.